Run jake in interactive mode so output isn't lost.

jbondc opened the pull request:
Fix jake perftsc.

msftclas commented:
Hi __@jbondc__, I'm your friendly neighborhood Microsoft Pull Request Bot (You can call me MSBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.microsoft.com.
    </span>

TTYL, MSBOT;

msftclas commented:
__@jbondc__, Thanks for signing the contribution license agreement so quickly! Actual humans will now validate the agreement and then evaluate the PR.
<br />Thanks, MSBOT;

DanielRosenwasser commented on tests/perfsys.ts:
```diff
@@ -54,22 +34,33 @@ module perftest {
+        }
+        try {
+            var log = <IOLog>JSON.parse(content);
+        } catch (err) {
```
catch on different line

DanielRosenwasser commented on tests/perfsys.ts:
```diff
@@ -20,29 +20,9 @@ module perftest {
     export var getCurrentDirectory = ts.sys.getCurrentDirectory;
     var exit = ts.sys.exit;
 
-    var args = ts.sys.args;
```
Can you explain why these changes were necessary? I'm personally not that familiar with the perfsys.ts stuff

jbondc replied:
There was a lot of hackery just to parse:
node .\built\local\perftsc.js --logio info.json b.ts
Then
node .\built\local\perftsc.js info.json

The args had to be 'reset' to be passed back to the compiler. As I was trying to fix it, didn't understand what it was trying to do. My understanding is you can benchmark the raw compile time, excluding any IO time spent fetching a file from the system.

DanielRosenwasser replied:
Perhaps @vladima can weigh in on this a bit.

DanielRosenwasser replied:
Perhaps @vladima can weigh in on this, but was this specifically something that was necessary to simplify the *Jakefile*?

jbondc replied:
Not necessary, I thought it would fix the travis failed build. Can't reproduce the error it's showing.

DanielRosenwasser commented:
Hey @jbondc, please remove the unnecessary changes and then try restoring the `stdout`/`stderr` listeners to the results of `createExec` calls. I don't see official documentation of an options parameter that takes `interactive` on `createExec` (but the Jake documentation is not very thorough at all).

If that works, please rebase and squash all the commits into one.

jbondc commented:
Should be working now, let me know if it looks good. Jake doesn't pipe stdout properly, something wrong with:
https://github.com/jakejs/jake/blob/master/lib/utils/index.js#L197
http://nodejs.org/api/child_process.html#child_process_options_stdio

There was also an issue with complete() inside exec() causing the builds to fail.

DanielRosenwasser commented on Jakefile:
```diff
@@ -303,19 +296,10 @@ compileFile(processDiagnosticMessagesJs,
-    });
-    ex.addListener("cmdEnd", function() {
+
+    exec(cmd, function() {
```
Why are these `function() { complete(); }` callbacks preferable to the old behavior?

jbondc replied:
Every {async: true} task needs a complete() callback.      

Just pushed another commit, hope its clearer and builds ok ;)

jbondc replied:
Went back to the old behavior, every callback now calls complete() so it works too. Renaming exec() to execAsync() and other functions that call complete().

DanielRosenwasser commented:
I asked a few members of the team; we don't seem to need `perfsys.ts` so just delete that file entirely. Then please undo the changes in `tsc.ts`

DanielRosenwasser commented on Jakefile:
```diff
@@ -252,15 +243,14 @@ function compileFile(outFile, sources, prereqs, prefixes, useBuiltCompiler, noOu
 
             if (callback) {
                 callback();
+            } else {
```
`else` on different line.

DanielRosenwasser commented on Jakefile:
```diff
@@ -475,22 +455,17 @@ var refTest262Baseline = path.join(internalTests, "baselines/test262/reference")
     ex.addListener("cmdEnd", function() {
         if (completeHandler) {
             completeHandler();
+        } else {
```
`else` on different line.

DanielRosenwasser commented on Jakefile:
```diff
@@ -233,17 +232,9 @@ function compileFile(outFile, sources, prereqs, prefixes, useBuiltCompiler, noOu
-        });
-        ex.addListener("cmdEnd", function() {
+        exec(cmd, function() {
+            console.log("")
```
What is this for?

jbondc replied:
console.log(cmd + "\n"); console.log("") Adds the extra "\n"

DanielRosenwasser commented on Jakefile:
```diff
@@ -475,22 +455,17 @@ var refTest262Baseline = path.join(internalTests, "baselines/test262/reference")
-        complete();
     });
-    ex.addListener("error", function(e, status) {
+    ex.addListener("error", errorHandler ? errorHandler : function(e, status) {
```
`errorHandler || function (e, status)`

DanielRosenwasser commented:
Sorry it took a bit to try it out. Looks good to me - let's give it a run and see how things work for us. Thanks @jbondc!

Arnavion commented:
This broke `jake local`. It stops after building servicesFile instead of continuing to build nodeDefinitionsFile and serverFile because the callback for servicesFile does not call complete()

Arnavion commented:
Also why tests are no longer running in travis when running `jake generate-code-coverage`. Compare https://travis-ci.org/Microsoft/TypeScript/builds/52949070 and https://travis-ci.org/Microsoft/TypeScript/builds/52950862

DanielRosenwasser commented:
I see, thanks for catching this @Arnavion. Reverting the merge. @jbondc feel free to send another PR based on the original branch with the appropriate fixes. I would prefer if you undid changes requiring the callback to call `complete()`, as calling it every time is clearly easy to forget.

jbondc commented:
Fixed, added a not that 'callback' must be synchronous:
https://github.com/jbondc/TypeScript/blob/contrib/Jakefile#L217

Restored the complete() on compileFile():
https://github.com/jbondc/TypeScript/blob/contrib/Jakefile#L272

Should I open a new pull request?
